### ISSUE TITLE
Setup Twitch Follower Webhook/Subscriber Websocket

### ISSUE THREAD

**Issue by adilanchian** [PixelogicDev/SimpleAlerts#8]
In order to receive information about a users follower/subscriber activtiy, we need to hook into one of Twitch's hooks to receive the data. At this point we should setup a API for bother subscribers and followers that will push updates to our client.

FLOW:
- I'm authenticated, I receive a new follower
- I receive payload through follower/new api
- Parse out the data I need, then send to client (Need to flesh out how to get to client)

**Comment by adilanchian** [comment]
UPDATE:
Subs and Bit cheers will require us to use PubSub
Followers will require use to use WebSub

**Comment by adilanchian** [comment]
Im testing a webhook here

**Comment by adilanchian** [comment]
Adding the docs here that would be EXTREMELY USEFUL:

https://dev.twitch.tv/docs

**Comment by adilanchian** [comment]
Alright, 

I finally figured out the issue with Twitch Follower Webhook. I was doing everything correctly, the issue was the localhost port that was being created for the callback. Unfortunately, it was not working and I believe it has to do with the amount of requests coming in with these massive streamers.

I switched to ngrok, changed route to `hook/follower/:id` and that seemed to work fantastically. Well, as long as I didn't get more than 20 connections a minute haha. Ngrok will time out at that point

Before I close this issue, I need to make sure to CLOSE THE CONNECTION when the user signs off or is not live anymore. I think the best bet is to check to see when the user is live or not. 

But ya, its working now woooo

**Comment by adilanchian** [comment]
I ended up adding in the Stream Up/Down Webhook. The flow of the app currently works as so:
- Sign in using Twitch Account
- Auth code sent to Server
- Server generates Auth Token
- Store this token in session storage
- Check to see if user is live when opening app, if so, setup follower webhook
- Setup stream up/down webhook
- On stream down, unsubscribe from follower webhook
- On stream up, subscribe to follower webhook

Currently the session storage is acting up so need to fix this. THEN I will move onto PubSub which should open sockets for Realtime subs and cheers!

**Comment by adilanchian** [comment]
Brought in ngrok module for easier local testing.

**Comment by adilanchian** [comment]
Starting implementing a session storage system on the server. Going to hold on this until I can #gitGud. This should be used to hold the auth token while the user is on the site.

**Comment by adilanchian** [comment]
Just implemented websockets here. No way to really just it unless I start getting an influx of subs/cheers. Reached out to Twitch Support to see if they recommended a way to test. Once testing is done, finalize clean up and finish session store!

**Comment by adilanchian** [comment]
Cleaning up PubSub stuff. Need to finish this issue off with session storage and mongodb storage.

**Comment by adilanchian** [comment]
I am going to put this on hold for now as I have taken a different approach... Using Streamlabs for everything at the moment. This definitely needs some code cleanup though before continuing.
